examples/geometric_art.rs

```diff
@@ -0,0 +1,127 @@
+    /// Create a new turtle using the given parameters.
+    /// `width`: total width of the drawing
+    /// `height`: total height of the drawing
+    /// `row_count`: number of rows of triangles to be created
+    /// `col_count`: number of columns per row to be created
+    /// `color_count`: number of different colors to use
+    fn new(width: u32, height: u32, row_count: u32, col_count: u32, color_count: u32) -> Self {
+        // Create and prepare the turtle.
+        let mut turtle = Turtle::new();
+        turtle.set_speed("instant");
+        turtle.pen_up();
+        turtle.drawing_mut().set_size([width, height]);
+
+        // Move turtle to the upper left corner of the drawing.
+        let size = turtle.drawing().size();
+        let x_start = size.width / 2;
+        let y_start = size.height / 2;
+        turtle.go_to([f64::from(x_start) * -1.0, f64::from(y_start)]);
+        turtle.set_speed(20);
+
+        // Calculate height and width of the triangles.
+        let row_height = f64::from(size.height) / f64::from(row_count);
+        let col_width = f64::from(size.width) / f64::from(col_count);
+
+        // Prepare a set of random colors to randomly choose from when drawing.
+        let mut colors = Vec::new();
+
+        for _i in 0..color_count {
+            colors.push(random::<Color>());
+        }
+
+        Self {
+            turtle,
+            row_count,
+            col_count,
+            row_height,
+            col_width,
+            colors,
+        }
+    }
+
+    /// Randomly choose a color from the previously specified set of colors.
+    fn get_color(&self) -> Color {
+        let index: usize = random_range(0, self.colors.len() - 1);
+        self.colors[index]
+    }
+
+    /// Draw a single triangle by drawing two sides and filling them with a random color
+    /// which creates the third side of the triangle along the way.
+    /// Note: Even though the turtle is always turning right in this method, this movement
+    /// can also result in a left turn (depending on the angle chosen).
+    fn draw_triangle(&mut self, angle: f64) {
+        self.turtle.set_fill_color(self.get_color());
+        self.turtle.begin_fill();
+
+        self.turtle.right(angle);
+        self.turtle.forward(self.col_width);
+
+        self.turtle.right(angle);
+        self.turtle.forward(self.row_height);
+
+        self.turtle.end_fill();
+    }
+
+    /// Draw a single row of triangles in two runs:
+    /// 1. Create triangles from left to right.
+    /// 2. Turn around and fill in triangles from right to left.
+    fn draw_row(&mut self) {
+        // Create an endless loop over the angles of 90 and 270 degrees
+        // (corresponds to moving right and left, respectively).
+        let angles = vec![90.0, 270.0];
+        let mut angle_cycle = angles.iter().cycle();
+
+        // Create triangles from left to right.
+        for _i in 0..self.col_count {
+            let angle = angle_cycle.next().unwrap();
```

Comment:
There's an opportunity here to use `angle_cycle.take(self.col_count)` instead of iterating over a range of numbers and calling `next()` + unwrapping. See the documentation for [take()](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.take) if you've never seen that before.

There's a similar loop below where this also applies.

Reply:
This took me some time to figure out how to do. 😅 At last, I managed to get it working. Let me know if I can improve this any further.